Mathlib/Analysis/SpecialFunctions/Complex/LogDeriv.lean

```diff
@@ -55,9 +55,11 @@ theorem hasStrictDerivAt_log {x : ℂ} (h : x ∈ slitPlane) : HasStrictDerivAt
 lemma hasDerivAt_log {z : ℂ} (hz : z ∈ slitPlane) : HasDerivAt log z⁻¹ z :=
   HasStrictDerivAt.hasDerivAt <| hasStrictDerivAt_log hz
 
+@[fun_prop]
```

Comment:
If you do the compositional lemmas such as `DifferentiableAt.clog`, I think these are superfluous (and - I believe - causes `fun_prop` to do extra work).

Reply:
`DifferentiableAt.clog` was exactly what I tagged to make `fun_prop` work in my use case, but it seems that you are right, and it suffices to tag the non-compositional lemmas.
Will remove the attribute from the others.

Reply:
@fpvandoorn per your comment to me on your recent `fun_prop` PR, we should actively *not* mark `differentiableAt_log` with the `fun_prop` tag for performance reasons, right? and the same for any of these "leaf" nodes, like `hasStrictFDerivAt_log_real`? If that's the case, then we need to remove all those `fun_prop` attributes from leaf nodes across the library.

I am confused about it though, because I would have expected that due to the discrimination tree indexing, these specific versions wouldn't be tried unless everything matched, and hence there would be essentially no performance cost.

Reply:
@MichaelStollBayreuth: I think `fun_prop` is actually quite good, and if you either tag `differentiable_log` or `Differentiable.log` it will succeed. @j-loreaux said that the latter was better to tag. I don't know if that is true, but maybe he has an example where it is better.
But my original message was intended to mean: *only* keep `fun_prop` for the compositional lemmas, not for the atomic ones, but my message was ambiguous.

However, if you tag both, `fun_prop` fails exponentially slowly: https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/fun_prop.20fails.20exponentially.20slowly
This problem is so pervasive in Mathlib, that tagging these two more lemmas won't really hurt until we have a proper solution for all of Mathlib.

Reply:
@fpvandoorn  Ah OK, I read "these" as referring to the compositional ones.
So I switch to tagging the compositional lemmas instead? (Or, if it is simpler, feel free to push to this branch if you like.)

Reply:
Yeah, keep only the compositional ones (or keep all of them, since `fun_prop` failing will take ~infinity time anyway).